src/View/Helper/FormSelect.php

```diff
@@ -205,7 +204,7 @@ public function renderOptions(array $options, array $selectedOptions = [])
                 $disabled = $optionSpec['disabled'];
             }
 
-            if (ArrayUtils::inArray($value, $selectedOptions)) {
+            if ($this->inArrayWithStringConversion($value, $selectedOptions)) {
```

Comment:
There is no need for a separate method, just use the `strict` variant: `ArrayUtils::inArray($value, $selectedOptions, true)`

Reply:
Unfortunatly, that won't work. Strict comparison failes because the unit tests assume that objects are identical when their string representation equals, but enabling strict comparison requires objects to be identical. Hence, if I replace the line as you suggested, the following unit tests fail:

```
1) LaminasTest\Form\View\Helper\FormSelectTest::testCanMarkOptionsAsSelectedWhenEmptyOptionOrZeroValueSelected
Failed asserting that '<select name="foo"><option value="">empty</option>\n
<option value="0">label0</option>\n
<option value="1">label1</option></select>' contains "<option value="0" selected="selected">label0</option>".

/home/driehle/PhpstormProjects/laminas-form/test/View/Helper/FormSelectTest.php:390

2) LaminasTest\Form\View\Helper\FormSelectTest::testRenderElementWithObjectIdentifiers
Failed asserting that '<select name="foo"><option value="42">This is the first label</option>\n
<option value="43">This is the second label</option></select>' matches PCRE pattern "#option .*?value="42" selected="selected"#".
```

To circumstance this, I have added the new method, which first converts objects to string and then performs a strict comparison. Such functionality is not provided in stdlib, I think.